Regression_test/RegressionSceneList.inl

```diff
@@ -106,6 +110,52 @@ void RegressionSceneList<T>::collectScenesFromList(const std::string& scenesDir,
             break;
     }
 
+    std::string fullPathReferenceDir = listDir + "/" + referencesDir;
+    // Check if the reference folder does exist
+    if (!referencesDir.empty())
+    {
+        if (!sofa::helper::system::FileSystem::exists(fullPathReferenceDir))
+        {
+            // relative reference path is wrong, check if the user set a env var instead
+            // most common case: regression references are not in the same namespace (aka repository)
+            char* refDirVar = getenv("REGRESSION_REFERENCES_DIR");
+            if (refDirVar == nullptr)
+            {
+                msg_error(msgHeader) << "The reference path does not exist, and the fallback REGRESSION_REFERENCES_DIR is not set.";
+                return;
+            }
+            else
+            {
+                std::string refDirFromVar(refDirVar);
+                // case where REGRESSION_REFERENCES_DIR is absolute
+                if (sofa::helper::system::FileSystem::isAbsolute(refDirFromVar) && sofa::helper::system::FileSystem::exists(refDirFromVar))
+                {
+                    fullPathReferenceDir = refDirFromVar;
+                }
+                else if(sofa::helper::system::FileSystem::exists(listDir + "/" + refDirFromVar))// case where REGRESSION_REFERENCES_DIR is relative
+                {
+                    fullPathReferenceDir = listDir + "/" + refDirFromVar;
+                }
+                else // all cases are not compliant, quitting
+                {
+                    msg_error(msgHeader) << "The reference path does not exist, and the fallback REGRESSION_REFERENCES_DIR is set but erroneous.";
+                    return;
+                }
+            }
+        }
+        else
```

Comment:
To be removed ?

Reply:
Otherwise this is great ! :+1:

Reply:
Removing the else () ; // OK ?
I like to always have a else() with a if(),  especially with the comment (kind of explicity way to say "this is where everything is ok")

Reply:
Ok, I understand. The compilo will remove it anyway, so if it increase readability that's ok !